cfpb/consumerfinance.gov
# TDP: semi-terrible fix for activities search bug

TDP has an existing bug in production 

<img width="298" alt="Screen Shot 2023-04-13 at 12 26 16 PM" src="https://user-images.githubusercontent.com/704760/231824226-38a25cc7-4a80-4c24-85b7-f41b7e891675.png">

This PR performs a semi-terrible solution of copying the older expandable JS code into TDP, in order to fix the bug, while isolating it from the current design-system expandable code.

## Additions

- Add a copy of the expandable template and JS into TDP.

## Changes

- Removes `is_collapsed_for_mobile` from standard expandable, which was added in https://github.com/cfpb/consumerfinance.gov/pull/6427 in service of TDP custom expandables.

## How to test this PR

1. `yarn build` and check http://localhost:8000/consumer-tools/educator-tools/youth-financial-education/teach/activities/?q= vs production and see that when the expandable facets are toggled they don't overlap themselves.


## Notes and todos

- The correct long-term solution for this is to develop out a component for the expandable-facets and the nested inputs, which then gets used in [TDP](https://www.consumerfinance.gov/consumer-tools/educator-tools/youth-financial-education/teach/activities/), [prepaid agreements](https://www.consumerfinance.gov/data-research/prepaid-accounts/search-agreements/), and [CCDB](https://www.consumerfinance.gov/data-research/consumer-complaints/search/). This work has already gotten underway in https://github.com/cfpb/design-system/pull/1628, however, the TDP nested checkboxes have the behavior of allowing checking of the children independent of checking the parent. So checking the parent does not equal checking all the children. Also, TDP uses link styling for the parents, while CCDB uses an underline-less hover that has inaccessible color contrast. These are issues that we don't want to enshrine in the design-system. So, the components should be built out in the DS in a proper way, and then TDP should be updated to use those new components.
